src/wp-includes/class-wp-theme-json.php

```diff
@@ -1925,6 +1925,53 @@ public function get_styles_block_nodes() {
 		return static::get_block_nodes( $this->theme_json );
 	}
 
+	/**
+	 * Returns a filtered declarations array if there is a separator block with only a background
+	 * style defined in theme.json by adding a color attribute to reflect the changes in the front.
+	 *
+	 * @since 6.1.1
+	 *
+	 * @param array $declarations List of declarations.
+	 *
+	 * @return array $declarations List of declarations filtered.
+	 */
+	private static function update_separator_declarations( $declarations ) {
+		$background_matches = array_values(
+			array_filter(
+				$declarations,
+				function( $declaration ) {
+					return 'background-color' === $declaration['name'];
+				}
+			)
+		);
```

Comment:
I committed something similar a few months back, and learned that `array_filter` goes over every item in the array.  Refactoring these as `foreach`s with a `break` after the first found instance is a nice little optimization.

Reply:
When using a closure as a callback, add `static` for a tiny performance micro-optimization:

```suggestion
					static function( $declaration ) {
```

Reply:
Hmm there are multiple `array_filter()`s iterating over the same dataset. This means the entire dataset is iterated each and every `array_filter()` in this method. 

Combining all of the `array_filter()` circuits into one iterator, such as `foreach()`, means the `$declarations`  array is walked 1 time instead of N times (N = each `array_filter()`). 

Granted, this means refactoring and re-testing the code to be more performant. Likely not something that can happen before 6.1.1-RC.

Reply:
@hellofromtonya I've refactored this to match your suggestions(at least as best as I understand them).

How does this look to you now?  Any further improvements we can make?

Reply:
@dream-encode I agree with @hellofromtonya, I think we can avoid using `array_filter()` here entirely and rather use a single `foreach`. Worth adding that also `array_values()` has a cost which we can avoid as well.

Given that #3555 is focused on optimizing performance by reducing excessive usage of array functions, it would feel counterproductive if we at the same time introduced new ones here.

IMO we can even have only a single `foreach` loop for the entire method, iterating through `$declarations` and checking for all three things (the first `background-color` value, whether there are `border-color` matches, and whether there are `color` matches).